app/tests/unit/middleware/requestSanitizer.spec.ts

```diff
@@ -0,0 +1,161 @@
+import express, { Request, Response } from 'express';
+import request from 'supertest';
+
+import { requestSanitizer } from '../../../src/middleware/requestSanitizer';
+
+function buildApp() {
+  const app = express();
+  // Body parsers
+  app.use(express.json());
+  app.use(express.urlencoded({ extended: true }));
+  app.use(requestSanitizer);
+
+  // Endpoints to inspect sanitized body
+  app.post('/echo', (req: Request, res: Response) => {
+    res.status(200).json(req.body);
+  });
+
+  app.get('/ping', (_req, res) => {
+    res.status(200).json({ ok: true });
+  });
+
+  return app;
+}
+
+describe('requestSanitizer middleware', () => {
+  let app: express.Express;
+  beforeEach(() => {
+    app = buildApp();
+  });
+
+  test('removes NUL written as \\0 (octal) inside JSON body', async () => {
```

Comment:
Style: While functionally equivalent, we should be using it(...) notation instead of test(...) notation.